### PR TITLE
SOLR-15862 Move solr/LICENSE.txt and SOLR/NOTICE.txt to git top-level

### PR DESCRIPTION
**Pull request by janhoy**
Adapt build to pick the license from there,
Adapt smokeTester to expect right set of files in root vs solr/

https://issues.apache.org/jira/browse/SOLR-15862

It is a requirement to have license and notice files on top-level. Like we did for Lucene-build I'm not moving these up. I tested with `gradlew dev` and confirmed that binary release works and contains the correct files.

### CODE REVIEW

File: gradle/java/jar-manifest.gradle [hunk 1]
```diff
@@ -22,7 +22,7 @@ subprojects {
     // per-project.
     def title = "Apache Solr Search Server: ${project.name}"
     def implementationTitle = "org.apache.solr"
-    def legaleseDir = project(":solr").projectDir
+    def legaleseDir = getRootProject().projectDir
```
Comment (dweiss) [inline gradle/java/jar-manifest.gradle:25]: just rootDir I think is enough.
https://docs.gradle.org/current/dsl/org.gradle.api.Project.html#org.gradle.api.Project:rootDir

File: solr/packaging/build.gradle [hunk 2]
```diff
@@ -92,13 +92,16 @@ distributions {
     contents {
       // TODO: This is missing module README files, as compared to 8x
 
+      from(rootProject.projectDir, {
```
Comment (dweiss) [inline solr/packaging/build.gradle:95]: Could be rootDir or this, fine.

**Review by dweiss — APPROVED** [review]

File: gradle/java/jar-manifest.gradle (hunk 1 above)
Comment (janhoy) [reply gradle/java/jar-manifest.gradle]: Done

File: solr/packaging/build.gradle (hunk 2 above)
Comment (janhoy) [reply solr/packaging/build.gradle]: Done

**Pull request merged**